### PR TITLE
feat: 🎸 add `usePrevious` hook with test

### PR DESCRIPTION
**Pull request by codyarose**
Ref #8

### CODE REVIEW

File: src/hooks/usePrevious.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {renderHook} from '@testing-library/react-hooks';
+import {usePrevious} from '.';
+
+const setup = ({initialValue}: {initialValue?: any} = {}) =>
```
Comment (SeanGroff) [inline src/hooks/usePrevious.test.ts]: Small `nit` but can we use `curly braces` and an explicit `return` whenever the return value falls to a new line? I believe this is somewhere in our Internal Code Standards

**Comment by SeanGroff** [comment]
@codyarose This looks REALLY solid. Could you add the documentation for this hook to our Readme? Doesn't need to be very elaborate. Something similar to what's seen for `useDialog` in the readme is sufficient.

File: src/hooks/usePrevious.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export function usePrevious<Type>(
```
Comment (SeanGroff) [inline src/hooks/usePrevious.ts:3]: Nice!

Do people still argue about `T` vs using something else for generics? `Type` reads more friendly IMO 🤷‍♂️

Comment (codyarose) [reply src/hooks/usePrevious.ts:3]: Yeah there was a bit of a commotion about that a while back. I think single letter generics make it more difficult to understand, especially for people who are newer to Typescript.

It's the same as using single letter variables/args, but with an extra layer or two of complexity

**Review by SeanGroff — APPROVED** [review]
Awesome 😎

**Pull request merged**